pages/_app.js

```diff
@@ -1,28 +1,37 @@
 import GlobalStyle from "../styles";
 import Image from "next/image";
-import styled from "styled-components";
+import plants from "/lib/data";
+import useLocalStorageState from "use-local-storage-state";
 
 export default function App({ Component, pageProps }) {
-  
+  const [allPlants, setAllPlants] = useLocalStorageState("allPlants", {
+    defaultValue: plants,
+  });
+
+  function handleToggleOwned(id) {
+    setAllPlants((prevPlants) =>
+      prevPlants.map((plant) =>
+        plant.id === id ? { ...plant, isOwned: !plant.isOwned } : plant
+      )
+    );
+  }
+
   return (
     <>
       <GlobalStyle />
-      <StyledHeader>
-        <Image 
+      <header>
```

Comment:
Consider using a [Layout component pattern](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts#layout-pattern) from Next.js to wrap the main structure of the page, including `<header>`, <`main>` and later the possible `<footer>` .

This approach can help with maintaining consistent styling 
and structure across different pages, making it easier to apply shared layout elements. It also improves code organization, as the layout logic is centralized.

**But that can be a task for the next print. 🏃**